messaging/src/test/java/org/axonframework/configuration/MessagingConfigurerTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.configuration;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import jakarta.annotation.Nullable;
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.SimpleCommandBus;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.commandhandling.gateway.DefaultCommandGateway;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventSink;
+import org.axonframework.eventhandling.SimpleEventBus;
+import org.axonframework.eventhandling.gateway.DefaultEventGateway;
+import org.axonframework.eventhandling.gateway.EventGateway;
+import org.axonframework.messaging.ClassBasedMessageTypeResolver;
+import org.axonframework.messaging.MessageTypeResolver;
+import org.axonframework.queryhandling.DefaultQueryGateway;
+import org.axonframework.queryhandling.QueryBus;
+import org.axonframework.queryhandling.QueryGateway;
+import org.axonframework.queryhandling.QueryUpdateEmitter;
+import org.axonframework.queryhandling.SimpleQueryBus;
+import org.axonframework.queryhandling.SimpleQueryUpdateEmitter;
+import org.junit.jupiter.api.*;
+
+/**
+ * Test class validating the {@link MessagingConfigurer}.
+ *
+ * @author Steven van Beelen
+ */
+class MessagingConfigurerTest extends ConfigurerTestSuite<MessagingConfigurer> {
```

Comment:
First, I like the test suites.
Two, I like how the hierarchy in configurers turned out for defaults. Nice :)